astropy/cosmology/flrw.py

```diff
@@ -1485,6 +1454,99 @@ def arcsec_per_kpc_proper(self, z):
         return u.arcsec / (self.angular_diameter_distance(z).to(u.kpc) *
                            arcsec_in_radians)
 
+    # -----------------------------------------------------
+
+    def __equiv__(self, other):
+        """FLRW equivalence. Use ``.is_equivalent()`` for actual check!
+
+        Parameters
+        ----------
+        other : `~astropy.cosmology.FLRW` subclass instance
+            The object in which to compare.
+
+        Returns
+        -------
+        `True` or `NotImplemented`
+            `True` if 'other' is of the same class and has matching parameters
+            and parameter values. `NotImplemented` otherwise.
+
+        """
+        # Note! this method is superseded by FlatFLRWMixin.__equiv__ in
+        # subclassses of FlatFLRWMixin.
+
+        if other.__class__ is not self.__class__:
+            return NotImplemented
+
+        # check all parameters in other match those in 'self' and 'other' has
+        # no extra parameters (latter part should never happen b/c same class)
+        sps, ops = self.info("parameters"), other.info("parameters")
+        params_eq = (all((np.all(ops[k] == v) for k, v in sps.items()))
+                     and not (sps.keys() ^ ops.keys()))
+
+        return params_eq or NotImplemented
+
+
+class FlatFLRWMixin(FlatCosmologyMixin):
+    """
+    Mixin class for flat FLRW cosmologies. Do NOT instantiate directly.
+    Must precede the base class in the multiple-inheritance so that this mixin's
+    ``__init__`` proceeds the base class'.
+    Note that all instances of ``FlatFLRWMixin`` are flat, but not all
+    flat cosmologies are instances of ``FlatFLRWMixin``. As example,
+    ``LambdaCDM`` **may** be flat (for the a specific set of parameter values),
+    but ``FlatLambdaCDM`` **will** be flat.
+    """
+
+    def __init__(self, *args, **kw):
+        super().__init__(*args, **kw)
+        # Do some twiddling after the fact to get flatness
+        self._Ode0 = 1.0 - self._Om0 - self._Ogamma0 - self._Onu0
+        self._Ok0 = 0.0
+
+    @property  # no longer a Parameter
+    def Ode0(self):
+        """Omega dark energy; dark energy density/critical density at z=0"""
+        return self._Ode0
```

Comment:
I'd normally have calculated this here, but I guess in this PR at least best to stick to how it is done elsewhere (and before); otherwise, would need to define `Ok0` as well...

Reply:
Yeah, especially since it's set in ``__init__`` for non-flat cosmologies.